[bug](build): restore OpenBLAS runtime for AArch64 ASan example tests

### Affected area
build

### Interface
cpp

### VSAG version
df9bb13e43dfbf0523da7f3562c76efd13c830aa

### Summary
The `Asan Build & Test Parallel` workflow fails in the `Test Example Aarch64` job before any example logic runs because the downloaded AArch64 example binaries cannot load `libopenblas.so.0` on the test runner.

The AArch64 build job installs dependencies with `scripts/deps/install_deps_ubuntu.sh`, which includes `libopenblas-dev`, then uploads the build artifact. The `Test Example Aarch64` job runs on a fresh `ubuntu-22.04-arm` runner, downloads the artifact, and executes examples without reinstalling the runtime dependency. In contrast, the AArch64 unit/function test job has a `Prepare Env` step before downloading and running tests.

### Minimal reproducer
Run the failing workflow job from GitHub Actions:

```shell
# Workflow: Asan Build & Test Parallel
# Run: 27181805164
# Job: Test Example Aarch64
cd ./build/examples/cpp
echo "leak:libgomp.so" > lsan.supp
echo "leak:libgomp.so.1" >> lsan.supp
export LSAN_OPTIONS=suppressions=$(pwd)/lsan.supp
for example in ./*; do
  filename=$(basename "$example")
  if [[ $filename =~ ^[0-9]{3} ]]; then
    chmod +x $example
    echo "Run $example"
    $example
  fi
done
```

Observed first example:

```shell
Run ./101_index_hnsw
./101_index_hnsw: error while loading shared libraries: libopenblas.so.0: cannot open shared object file: No such file or directory
```

### Environment
GitHub Actions failed job environment:

```shell
Workflow: Asan Build & Test Parallel
Run URL: https://github.com/antgroup/vsag/actions/runs/27181805164
Job URL: https://github.com/antgroup/vsag/actions/runs/27181805164/job/80243079328
Job: Test Example Aarch64
Runner image: ubuntu-22.04-arm
Operating system: Ubuntu 22.04.5 LTS
Runner version: 2.334.0
Image version: 20260531.15.1
Head SHA: df9bb13e43dfbf0523da7f3562c76efd13c830aa
```

Local `bash scripts/check_environment.sh` output from issue drafting:

```shell
- OS: Ubuntu 22.04
- vsag version: v0.18.0-267-gdf9bb13e
- compiler version: c++ (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0
```

### Expected behavior
`Test Example Aarch64` should install or otherwise provide the OpenBLAS runtime needed by the downloaded example binaries, then execute all numbered C++ examples.

### Actual behavior
The job fails immediately on `./101_index_hnsw` with exit code 127:

```shell
./101_index_hnsw: error while loading shared libraries: libopenblas.so.0: cannot open shared object file: No such file or directory
##[error]Process completed with exit code 127.
```

### References
- https://github.com/antgroup/vsag/actions/runs/27181805164/job/80243079328
- `.github/workflows/asan_build_and_test.yml:49` builds AArch64 and installs dependencies before artifact upload.
- `.github/workflows/asan_build_and_test.yml:121` AArch64 unit/function tests reinstall dependencies before running downloaded binaries.
- `.github/workflows/asan_build_and_test.yml:194` AArch64 example tests download and run binaries without a `Prepare Env` dependency install step.
- `scripts/deps/install_deps_ubuntu.sh:2` includes `libopenblas-dev` in the shared Ubuntu dependency list.

### Additional context
The x86 example job runs inside a CI container image, while the AArch64 example job runs directly on `ubuntu-22.04-arm`. Adding the same dependency preparation used by `test_asan_aarch64`, or otherwise ensuring `libopenblas.so.0` is available in the AArch64 example test environment, should address this specific failure.

## Related issues
_None found._

---
_Drafted with AI assistant: OpenCode:gpt-5.5_

--- a/.github/workflows/asan_build_and_test.yml
+++ b/.github/workflows/asan_build_and_test.yml
@@ -202,6 +202,8 @@ jobs:
       - name: Free Disk Space
         run: rm -rf /opt/hostedtoolcache
       - uses: actions/checkout@v4
+      - name: Prepare Env
+        run: sudo bash ./scripts/deps/install_deps_ubuntu.sh
       - name: Clean Env
         run: rm -rf ./build
       - name: Download Test

